src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.cs

```diff
@@ -63,14 +64,30 @@ internal SdkDirectoryWorkloadManifestProvider(string sdkRootPath, string sdkVers
             }
 
             _sdkRootPath = sdkRootPath;
+            _sdkOrUserLocalPath = sdkRootPath;
             _sdkVersionBand = new SdkFeatureBand(sdkVersion);
             _workloadSetVersionFromConstructor = workloadSetVersion;
             _globalJsonPathFromConstructor = globalJsonPath;
 
-            var knownManifestIdsFilePath = Path.Combine(_sdkRootPath, "sdk", sdkVersion, "KnownWorkloadManifests.txt");
+            string? userManifestsRoot = userProfileDir is null ? null : Path.Combine(userProfileDir, "sdk-manifests");
+            string dotnetManifestRoot = Path.Combine(_sdkOrUserLocalPath, "sdk-manifests");
```

Comment:
If I'm following the code correctly, at this point there is no difference between `_sdkRootPath` and `_sdkOrUserLocalPath`, and `_sdkOrUserLocalPath` may change later.  So I think it would be clearer to use the value that doesn't change:

```suggestion
            string dotnetManifestRoot = Path.Combine(_sdkRootPath, "sdk-manifests");
```

It might also be clearer to move the initialization of `_sdkOrUserLocalPath` closer to where it may be set to the user local path, or to set it to null initially and then set it to `_sdkRootPath` after the following if statement.